### DISCUSSION TITLE
Lvgl7 version supportet?

### DISCUSSION THREAD

**Discussion by NeroBurner** [bobwolff68/LVGLPlusPlus#3] [General]
Hi. I'd like to port https://github.com/InfiniTimeOrg/InfiniTime from lvgl7 to lvgl8. And maybe in the process allow a compile time switch between 7 and 8 for the porting effort and to compare RAM usage as the PineTime smartwatch is very memory constrained.

So I'd like to ask if lvgl++ is able to work with lvgl7 and lvgl8 and provide a layer of abstraction

**Comment by bobwolff68** [comment]
Hi there. The LVGLPlusPlus library was developed against version 8.2 of LVGL I believe. I figured I had that in the documentation, but in looking around I don't see record of it. I believe version 7, 8, and 9 have pretty considerable API / compatibility changes, but I would say you should go to the lvgl forum https://forum.lvgl.io/ and do a bit of searching for version compatibility. I'm sure it was a talked about discussion pretty frequently as this is a pretty typical question to ask and be concerned with.

Good luck on your project and let me know what you find out.
